### ISSUE TITLE
fn:hash, CRC-32: Describe output

### ISSUE THREAD

**Issue by ChristianGruen** [qt4cg/qtspecs#1214]
In the [QT4 CG Meeting 077](https://qt4cg.org/meeting/minutes/2024/05-14.html) it was suggested that the binary output of the newly added `CRC-32` algorithm should be further described.

**Comment by michaelhkay** [comment]
This led me to try and find the specification of CRC-32. The current link to IEEE 802-3 doesn't seem very helpful, can we be more specific?

Generally these algorithms specify that the hash value is an n-bit quantity, so I think mapping that to a hexBinary value isn't a process that requires much explanation. I think some of the confusion might be because the examples are showing the result after conversion to a string - which, of course, is the only way you can show them.

**Comment by ChristianGruen** [comment]
> This led me to try and find the specification of CRC-32. The current link to IEEE 802-3 doesn't seem very helpful, can we be more specific?

I thought it was obvious enough, but I’ll be glad to replace the link with a more appropriate one.

**Comment by ndw** [comment]
When XProc added CRC32 to the `p:hash` step, we referenced:

> [CRC32] “32-Bit Cyclic Redundancy Codes for Internet Applications”, The International Conference on Dependable Systems and Networks: 459. 10.1109/DSN.2002.1028931. P. Koopman. June 2002. 

Which is available online in [this PDF](https://users.ece.cmu.edu/~koopman/networks/dsn02/dsn02_koopman.pdf), I believe.

The actual implementation I use in XML Calabash is `java.util.zip.CRC32`.

**Comment by ChristianGruen** [comment]
> I think some of the confusion might be because the examples are showing the result after conversion to a string - which, of course, is the only way you can show them.

Would the spec rules allow is us to present the result as `xs:hexBinary('01234567')`?

**Comment by michaelhkay** [comment]
I think it would be a lot less confusing if instead of

```
**Expression:**	
hash("abc") => string()
**Result:**	
"900150983CD24FB0D6963F7D28E17F72"

```
we wrote:

```
**Expression:**	
hash("abc") 
**Result:**	
xs:hexBinary("900150983CD24FB0D6963F7D28E17F72")
```

**Comment by ChristianGruen** [comment]
@rhdunn Do you think this issue can be marked as resolved with the revised examples (#1222)?

**Comment by rhdunn** [comment]
Yes.
